src/power_grid_model/core/power_grid_dataset.py

```diff
@@ -266,6 +273,18 @@ def _validate_properties(self, data: ComponentData, schema: ComponentMetaData):
         if properties.batch_size != self._batch_size:
             raise ValueError(f"Dataset must have a consistent batch size across all components. {VALIDATOR_MSG}")
 
+    def _validate_dtypes_compatibility(self, data: ComponentData, schema: ComponentMetaData):
+        sub_data = _extract_data_from_component_data(data)
+        if is_columnar(data):
+            for attr, array in sub_data.items():
+                if schema.dtype.names is None or attr not in schema.dtype.names:
+                    raise Warning(f"Attribute {attr} is not in schema. {VALIDATOR_MSG}")
+                if array.dtype != schema.dtype[attr]:
+                    raise Warning(f"Data type for attribute {attr} does not match schema. {VALIDATOR_MSG}")
+        else:
+            if sub_data.dtype != schema.dtype:
+                raise Warning("Data type does not match schema. {VALIDATOR_MSG}")
+
```

Comment:
this is the worst of both worlds because the user will get an exception (as in the sustainable check) but it's not the exception they expect.

Please have a look at the steps mentioned in #774 (especially about the existing solution used in the deprecated `import/export_*` functionality in `utils`. In particular, use `warnings.warn` with the correct category, rather than `raise` manually